Eslint autofix for 2.x

bajtos opened the pull request:
@Amir-61 I run `eslint --fix` on 2.x code base to fix at least some of the differences between `master` and `2.x`. Will this make it easier for you to back-port changes from master to 2.x?

I'll leave it up to you to decide whether to land this pull request or close it unmerged.

superkhau commented:
LGTM, but amir for final check

Amir-61 commented on lib/connectors/memory.js:
```diff
@@ -567,9 +567,9 @@ function applyFilter(filter) {
-                 testInEquality({lte:example.between[1]}, value) );
+      if ('between' in example) {
+        return (testInEquality({ gte:example.between[0] }, value) &&
+                 testInEquality({ lte:example.between[1] }, value));
```
Just out of curiosity if we break a line to multiple lines, what is the preferred style?

Based on `Eslint --fix` it seems it adds 9 spaces to indent the second line; should it be two spaces?
```
return (testInEquality({ gte:example.between[0] }, value) &&
  testInEquality({ lte:example.between[1] }, value));
```

Just wanted to know the preference in the team; also the proceeding lines do not have any more indentation, I believe, right?

Example:
```
return (testInEquality({ gte:example.between[0] }, value) &&
  testInEquality({ lte:example.between[1] }, value) &&
  testInEquality({ lte:example.between[2] }, value));
```

superkhau replied:
@bajtos can field this one. Please update the style-guide too with this answer as I'm also curious.

bajtos replied:
Good catch! Apparently eslint doesn't care whether its 2 spaces, 8 spaces or 9 spaces in this case.

Your example looks exactly as what I would write myself, @Amir-61:
>```js
return (testInEquality({ gte:example.between[0] }, value) &&
  testInEquality({ lte:example.between[1] }, value) &&
  testInEquality({ lte:example.between[2] }, value));
```

bajtos replied:
> Please update the style-guide too with this answer as I'm also curious.

https://github.com/strongloop-internal/loopback-knowledge-base/pull/10

Amir-61 commented:
@slnode test please

Amir-61 commented:
LGTM

bajtos commented:
I have rebased on top of the current 2.x. While I was addressing Amir's comment, I fixed all eslint issues I found in the memory connector (see the second commit).

bajtos commented:
@Amir-61 PTAL one more time

Amir-61 commented:
LGTM
